src/mame/drivers/blit.cpp

```diff
@@ -0,0 +1,312 @@
+	// octal 0027 - ???
+	// octal 0030 - 16-bit
+	map(0x060018, 0x060019).w(FUNC(blit_state::start_write));
+	// octal 0040 - 16-bit
+	map(0x060020, 0x060029).ram().share("misccr");
+	// octal 0050 - 16-bit
+//	map(0x060028, 0x060029).w(FUNC(blit_state::init_write));
+	// octal 0060, 0062 - 8-bit - keyboard
+	map(0x060030, 0x060031).rw(m_acia1, FUNC(acia6850_device::status_r), FUNC(acia6850_device::control_w));
+	map(0x060032, 0x060033).rw(m_acia1, FUNC(acia6850_device::data_r), FUNC(acia6850_device::data_w));
+	// octal 0070 - 16-bit
+	map(0x060038, 0x060039).w(FUNC(blit_state::vblank_ack));
+}
+
+static INPUT_PORTS_START(blit)
+INPUT_PORTS_END
+
+static DEVICE_INPUT_DEFAULTS_START( host_rs232_defaults )
+	DEVICE_INPUT_DEFAULTS( "RS232_TXBAUD", 0xff, RS232_BAUD_19200 )
+	DEVICE_INPUT_DEFAULTS( "RS232_RXBAUD", 0xff, RS232_BAUD_19200 )
+	DEVICE_INPUT_DEFAULTS( "RS232_STARTBITS", 0xff, RS232_STARTBITS_1 )
+	DEVICE_INPUT_DEFAULTS( "RS232_DATABITS", 0xff, RS232_DATABITS_8 )
+	DEVICE_INPUT_DEFAULTS( "RS232_PARITY", 0xff, RS232_PARITY_NONE )
+	DEVICE_INPUT_DEFAULTS( "RS232_STOPBITS", 0xff, RS232_STOPBITS_2 )
+	DEVICE_INPUT_DEFAULTS( "FLOW_CONTROL", 0x01, 0x01 )
+DEVICE_INPUT_DEFAULTS_END
+
+static DEVICE_INPUT_DEFAULTS_START( kbd_rs232_defaults )
+	DEVICE_INPUT_DEFAULTS( "RS232_TXBAUD", 0xff, RS232_BAUD_19200 )
+	DEVICE_INPUT_DEFAULTS( "RS232_RXBAUD", 0xff, RS232_BAUD_19200 )
+	DEVICE_INPUT_DEFAULTS( "RS232_STARTBITS", 0xff, RS232_STARTBITS_1 )
+	DEVICE_INPUT_DEFAULTS( "RS232_DATABITS", 0xff, RS232_DATABITS_8 )
+	DEVICE_INPUT_DEFAULTS( "RS232_PARITY", 0xff, RS232_PARITY_NONE )
+	DEVICE_INPUT_DEFAULTS( "RS232_STOPBITS", 0xff, RS232_STOPBITS_2 )
+	DEVICE_INPUT_DEFAULTS( "FLOW_CONTROL", 0x01, 0x01 )
+DEVICE_INPUT_DEFAULTS_END
+
+
+void blit_state::start_write(offs_t offset, uint16_t data, uint16_t mem_mask)
+{
+	LOG("START <- %04X (addr %06X)\n", data, data << 2);
+	m_videostart = data << 2;
+	m_videoram = (uint8_t *)m_maincpu->space(AS_PROGRAM).get_read_ptr(m_videostart);
+	assert(m_videoram != NULL);
+}
+
+WRITE_LINE_MEMBER(blit_state::system_clock_write)
+{
+	m_acia0->write_txc(state);
+	m_acia0->write_rxc(state);
+	//
+	m_acia1->write_txc(state);
+	m_acia1->write_rxc(state);
+}
+
+/* Boot vector handler, the PCB hardwires the first 8 bytes from 0xff800000 to 0x0 at reset */
+uint16_t blit_state::bootvect_r(offs_t offset)
+{
+	return m_sysram_on ? m_sysram[offset] : m_sysrom[offset];
+}
+
+void blit_state::bootvect_w(offs_t offset, uint16_t data, uint16_t mem_mask)
+{
+	m_sysram[offset % sizeof(m_sysram)] &= ~mem_mask;
+	m_sysram[offset % sizeof(m_sysram)] |= (data & mem_mask);
+	m_sysram_on = true;
+}
+
+void blit_state::vblank_ack(offs_t offset, uint16_t data, uint16_t mem_mask)
+{
+	m_maincpu->set_input_line(M68K_IRQ_1, CLEAR_LINE);
+}
+
+
+uint32_t blit_state::screen_update(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect)
+{
+	uint8_t gfx = 0;
+	int x, y, z, fg, bg;
```

Comment:
Please scope down the `gfx`, `x`, `y` and `z` variables.

Reply:
ok.